Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -439,16 +446,23 @@ protected virtual GrpcChannelOptions GetChannelOptions()
 
         /// <summary>
         /// Returns whether or not a channel pool can be used if a channel is required. The default behavior is to return
-        /// true if and only if no quota project, scopes, credentials or token access method have been specified. Derived classes should
-        /// override this property if there are other reasons why the channel pool should not be used.
+        /// true if and only if no quota project, scopes, credentials or token access method have been specified and 
+        /// if UseJwtAccessWithScopes flag matches the flag in ChannelPool. 
+        /// Derived classes should override this property if there are other reasons why the channel pool should not be used.
         /// </summary>
         protected virtual bool CanUseChannelPool =>
             ChannelCredentials == null &&
             CredentialsPath == null &&
             JsonCredentials == null &&
             TokenAccessMethod == null &&
             Scopes == null &&
-            QuotaProject == null;
+            QuotaProject == null &&
+            UseJwtAccessWithScopes == GetChannelPool().UseJwtAccessWithScopes;
```

Comment:
Note to self (not a request for changes): some APIs make GetChannelPool throw an exception, but that's okay because they also override CanUseChannelPool.

Reply:
Yes, I think it's fine to assume that GetChannelPool is "the one we've provided" and that we won't do anything weird.